### DISCUSSION TITLE
Use RS485 instead of LIN transceiver

### DISCUSSION THREAD

**Discussion by Anoxym** [JanM321/esphome-lg-controller#87] [General]
Hi,

I'm a bit confused about the protocols used to control the LG system.

I already have a module called LG-RC-MBS-1 that controls the LG system via RS485.

![image](https://github.com/user-attachments/assets/8bae6b95-ece8-48d4-9d3a-80d1a5a7f024)

Could anyone help me or guide me on how to modify the code to use this module instead of the LIN transceiver?
I assume I just need to update the connection part?

Here’s the module manual, which includes a table listing the register addresses for controlling the LG system:

[LG-RC-MBS-1 User Manual](https://cdn.chipkin.com/assets/uploads/2018/feb/16-01-14-46_intesisbox_lg-rc-mbs-1_user_manual_en.pdf)

I have an ESP32 with an RS485 output that will be connected to the RS485 module via A+ and B- and B Y R to LG unit.

Can someone guide me?

**Comment by JanM321** [comment]
I don't think that's possible? The ESPHome controller in this repository talks to the indoor unit using LG's B/Y/R connection/protocol. The IntesisBox module does the same thing, but it also offers RS485. This is very different from the LG protocol though - it seems to use Modbus.

**Comment by Anoxym** [comment]
The intensisBox is wired to indoor unit too, so the best way is to replace my RS485 on esp32 and add a LIN transceiver ? then :
B : GND
Y : DATA
R : +12V
I don't need the +12V, just need to wire B and Y to LIN ?
Actually, I want to use the IntesisBox because it would save me from having to buy the LIN module, as I already have it and it works with the current system. Besides, it is specifically designed to control the LG environment.

**Comment by KrzysztofHajdamowicz** [comment]
LIN Transceiver is included to provide safe electrical framework for interfacing with LG indoor unit.
Physically it's 104bps serial connection over 12V signal.
